doc/adr/0001-jetstream-json-api-design.md

```diff
@@ -0,0 +1,131 @@
+# 1. JetStream JSON API Design
+
+Date: 2020-04-30
+
+## Status
+
+Proposed
+
+## Context
+
+At present, the API encoding consists of mixed text and JSON, we should improve consistency and error handling.
+
+## Decision
+
+We anticipate many types of consumer for JetStream including very small devices via MQTT and those written in 
+languages without easy to use JSON support like C.
+
+We will make it easy for light-weight clients by using `+OK` and `-ERR` style prefixes for client interactions,
+but the server management API will be JSON all the time. The rationale being that such small devices will most
+likely have their Consumers and Streams provisioned as part of the platform and often might not care for the
+additional context.
+
+In client responses we will use URL Encoding to provide additional context so multiple fields worth of
+context can be provided. URL Encoding has [broad language support](https://www.rosettacode.org/wiki/URL_decoding).
+
+### Admin APIs
+
+#### Requests
+
+All Admin APIs that today accept `nil` body should also accept an empty JSON document as request body.
+
+#### Responses
+
+All responses will be JSON objects, a few examples will describe it best. Any error that happens has to be 
+communicated within the originally expected message type. Even the case where JetStream is not enabled for
+an account, the response has to be the data type the API would normally generate with additional `error`.
+
+Successful Stream Info:
+
+```json
+{
+  "type": "io.nats.jetstream.api.v1.stream_info",
+  "time": "2020-04-23T16:51:18.516363Z",
+  "config": {
+    "name": "STREAM",
+    "subjects": [
+      "js.in"
+    ],
+    "retention": "limits",
+    "max_consumers": -1,
+    "max_msgs": -1,
+    "max_bytes": -1,
+    "max_age": 31536000,
```

Comment:
@derekcollison reminder to make these seconds

Reply:
for fields like `max_age`, attaching the unit of the field provides a reminder of what is expected, rather than requiring a doc expedition.

Reply:
Possibly, but I'll park this for another time so marking as resolved, dont want another side track